renormalizer/mps/matrix.py

```diff
@@ -46,7 +46,10 @@ def __init__(self, array, dtype=None, is_full_mpdm=False):
         backend.running = True
 
     def __getattr__(self, item):
-        res = getattr(self.array, item)
+        try:
```

Comment:
Could you explain why this necessary?

Reply:
It is noticed that in the `__init__` function inside `class matrix` exist `self.array`, however, once `self` has no attribute `array`, it calls `__getattr__`, inside which undergoes `getattr(self.array, item)`, Which brings to an endless loop between them, and finally lead to a maximum recursion number. I met such problems when attempting to enable *multiprocessing*, upon which I have discussed with you before.

Reply:
Forgive me but I still can't understand this. I remember the discussion but not the details. I've got two questions:
- When will an instance of `Matrix` be accessed without the attribute `array`. Is it during multiprocessing? 
- How does the current code fix this? Will a maximum recursion error be raised and caught by the `except` clause? Leading to an `AttributeError`?

Reply:
- Yes, such circumstance occurs only when using multiprocessing. I added `print(item)` after `except`, and I got a lot of `array` printed.
- I refered to [https://www.cnblogs.com/blackmatrix/p/5681480.html], then I found it worked! Still, I need further study to understand it.

Reply:
Here's what happens. When starting multiprocessing empty `Matrix`s are *pickled*, leading to the infinite recursion. Your clause will catch the maximum recursion error and turn it into an `AttributeError`. The error is propagated and caught by the `pickle` module which feels OK with `AttributeError`. So everything is working as expected.

I've improved your approach in the lastest commit in two ways:
- More documents on what happens.
- Raise `AttributeError` directly instead of finishing 500 times of recursion firstly. This should be beneficial to the performance.